datarobot_provider/operators/custom_job.py

```diff
@@ -0,0 +1,462 @@
+# Copyright 2023 DataRobot, Inc. and its affiliates.
+#
+# All rights reserved.
+#
+# This is proprietary source code of DataRobot, Inc. and its affiliates.
+#
+# Released under the terms of DataRobot Tool and Utility Agreement.
+import json
+import os
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import Iterable
+
+import datarobot as dr
+from airflow.exceptions import AirflowException, AirflowFailException
+from airflow.models import BaseOperator
+
+from datarobot_provider.hooks.datarobot import DataRobotHook
+
+
+class CreateCustomJobOperator(BaseOperator):
+    """
+    Create a DataRobot custom job.
+    :param name: custom job name
+    :type name: str
+    :param description: custom job description
+    :type description: str, optional
+    :return: custom job ID
+    :rtype: str
+    """
+
+    # Specify the arguments that are allowed to parse with jinja templating
+    template_fields: Iterable[str] = [
+        "name",
+        "description",
+    ]
+    template_fields_renderers: Dict[str, str] = {}
+    template_ext: Iterable[str] = ()
+    ui_color = '#f4a460'
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        description: str = None,
+        datarobot_conn_id: str = "datarobot_default",
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.name = name
+        self.description = description
+        self.datarobot_conn_id = datarobot_conn_id
+        if kwargs.get('xcom_push') is not None:
+            raise AirflowException(
+                "'xcom_push' was deprecated, use 'BaseOperator.do_xcom_push' instead"
+            )
+
+    def execute(self, context: Dict[str, Any]) -> str:
+        # Initialize DataRobot client
+        DataRobotHook(datarobot_conn_id=self.datarobot_conn_id).run()
+
+        if self.name is None:
+            raise ValueError("Invalid or missing custom job name")
+
+        response = dr.client.get_client().post(
+            "customJobs/",
+            data={
+                "name": self.name,
+                "description": self.description,
+            },
+        )
```

Comment:
With Custom Jobs going GA in 10.0 and having support in the Python Client at that time, I think we should postpone this PR until that time.